torch/autograd/functional.py

```diff
@@ -602,7 +690,7 @@ def vjp(grad_output):
 
         return _tuple_postprocess(jacobian, (is_outputs_tuple, is_inputs_tuple))
 
-def hessian(func, inputs, create_graph=False, strict=False, vectorize=False):
+def hessian(func, inputs, create_graph=False, strict=False, vectorize=False, forward_ad=False):
```

Comment:
We probably want some way to specify that the inner jacobian should happen via reverse-mode AD and the outer jacobian should happen with forward-mode AD. If the function is some R^N -> R^M with N > M, then that way is the best for performance (we use reverse-mode to compute jacobians of R^N -> R^M and then use forward mode to compute a jacobian of the resulting R^N -> R^{M + N} function)